utils/src/main/java/org/opentripplanner/utils/time/DurationUtils.java

```diff
@@ -123,6 +124,49 @@ public static Duration duration(String duration) {
     }
   }
 
+  /**
+   * Parse a duration string in format hh:mm:ss.
+   * @param duration string in format hh:mm:ss
+   * @return Duration
+   * @throws DateTimeParseException on bad input
+   */
+  public static Duration parseClockDuration(String duration) {
```

Comment:
I wonder if we need this or could we just use `Duration.between(LocalTime.MIN, LocalTime.parse(duration))`? The `parse` function probably accepts more formats that what is in theory allowed by OSM tagging.

Reply:
There are also factories in the data API to create custom parsers - we do not need this.

Reply:
Joel also wanted to use Duration.between(LocalTime.MIN, LocalTime.parse(duration, DateTimeFormatter.ISO_LOCAL_TIME)), which I guess will probably work ok for durations smaller than 24 hours. But the format specified in https://wiki.openstreetmap.org/wiki/Key:duration does not limit the duration to under 24 hours, which is of course irrelevant for the escalator use case, but if we ever want to recycle the duration tag parsing code for other durations, it might become relevant. Let's go through this in today's meeting.

Reply:
But now I noticed that my implementation is against the spec anyway. It specifies that the duration is either mm, hh:mm or hh:mm:ss. Not hh, hh:mm or hh:mm:ss as I implemented it.